pkg/didmethod/peer/generate.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package peer
+
+import (
+	"crypto"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+)
+
+const (
+	//numAlgo is the algorithm for choosing a numeric basis Reference: https://openssi.github.io/peer-did-method-spec/index.html#method-specific-identifier
+	numAlgo = "1"
+	//encAlgo  is the algorithm for encoding
+	encAlgo = "1"
+)
+
+//NewDid creates the peer DID. For example : did:peer:11-479cbc07c3f991725836a3aa2a581ca2029198aa420b9d99bc0e131d9f3e2cbe
+func NewDid(doc *did.Doc) (string, error) {
```

Comment:
@talwinder50 it would be better if the DID is an attribute of the `did.Doc`. As in, `did := doc.DID()` where `doc` is a peer DID document

Basically, all DID documents and their attributes should be under a single unified model

Reply:
@rolsonquadras @llorllale when you say peer document you mean genesis document ?

Reply:
@talwinder50 genesis doc(version) is the first version of peer doc.

Reply:
done